### PR TITLE
API Endpoint and functionality for Centrality calculations

### PR DESCRIPTION
**Pull request by koebi**
### Pull Request Checklist

- [X] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the master branch into my feature branch and all conflicts have been resolved.
- [X] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [X] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [X] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [X] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [X] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [X] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [X] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [x] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [X] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

### Information about the changes
- Key functionality added:
Implement API endpoint `centrality` and functionality to calculate betweenness centrality for nodes in a given bounding box.

### CODE REVIEW

File: openrouteservice/src/main/java/org/heigit/ors/api/responses/centrality/CentralityResponse.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    @JsonIgnore
+    protected BoundingBox bbox;
+
+    @JsonIgnore
+    protected CentralityResult centralityResults;
+
+    public CentralityResponse() {};
+
+    // In RouteResponse, this method was used to get metadata from RouteRequest.
+    // The unused CentralityRequest is a remnant of this.
```
Comment (sfendrich) [inline openrouteservice/src/main/java/org/heigit/ors/api/responses/centrality/CentralityResponse.java]: If it is not necessary, then please remove the parameter

File: openrouteservice/src/main/java/org/heigit/ors/centrality/algorithms/brandes/BrandesCentralityAlgorithm.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        }
+
+
+        for (int s : nodesInBBox) {
+            Stack<Integer> S = new Stack<>();
+            HashMap<Integer, List<Integer>> P = new HashMap<>();
+            HashMap<Integer, Integer> sigma = new HashMap<>();
+
+            // single source shortest path
+            //S, P, sigma = SingleSourceDijkstra(graph, nodesInBBox, s);
```
Comment (sfendrich) [inline openrouteservice/src/main/java/org/heigit/ors/centrality/algorithms/brandes/BrandesCentralityAlgorithm.java:53]: This is just a question, nothing to be done: could one of the existing routing algorithms be used here?

File: openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfile.java [hunk 3]
```diff
@@ -660,6 +665,55 @@ public MatrixResult computeMatrix(MatrixRequest req) throws Exception {
...
+                }
+            }
+        });
+
+        CentralityAlgorithm alg = new BrandesCentralityAlgorithm();
+        alg.init(graph, weighting);
+
+        HashMap<Integer, Double> betweenness = alg.compute(nodesInBBox);
+
+        //System.out.printf("Working on %d nodes, result has %d nodes\n, nodesInBBox.size(), betweenness.size());
```
Comment (sfendrich) [inline openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfile.java]: Please remove commented out code. If you need such an output for debugging, consider using an appropriate logger.

**Review by sfendrich — CHANGES_REQUESTED** [review]
Please also remove unused imports.

File: openrouteservice/src/main/java/org/heigit/ors/centrality/algorithms/brandes/BrandesCentralityAlgorithm.java (hunk 2 above)
Comment (koebi) [reply openrouteservice/src/main/java/org/heigit/ors/centrality/algorithms/brandes/BrandesCentralityAlgorithm.java:53]: Not without any modification - as far as I know, no routing algorithm keeps track of predecessors on shortest paths or even considers alternative shortest paths of same length.

**Review by sfendrich — APPROVED** [review]

**Pull request merged**